tests/unittests/Foundation/NSStringTests.m

```diff
@@ -739,22 +744,32 @@ void testUrlCharacterSetEncoding(NSString* decodedString, NSString* encodedStrin
 }
 
 TEST(NSString, LocalizedCaseInsensitiveContainsString) {
+    EXPECT_FALSE([@"" localizedCaseInsensitiveContainsString:nil]);
     EXPECT_FALSE([@"" localizedCaseInsensitiveContainsString:@""]);
     EXPECT_FALSE([@"TÉST" localizedCaseInsensitiveContainsString:@""]);
     EXPECT_TRUE([@"TÉST" localizedCaseInsensitiveContainsString:@"T"]);
     EXPECT_TRUE([@"TÉST" localizedCaseInsensitiveContainsString:@"TÉST"]);
     EXPECT_TRUE([@"TÉST" localizedCaseInsensitiveContainsString:@"tést"]);
+    EXPECT_FALSE([@"TE\u0301ST" localizedCaseInsensitiveContainsString:@"TÉST"]);
+    EXPECT_FALSE([@"TÉST" localizedCaseInsensitiveContainsString:@"TE\u0301ST"]);
+    EXPECT_TRUE([@"اختبار النص" localizedCaseInsensitiveContainsString:@"اختبار النص"]);
+    EXPECT_TRUE([@"اختبار النص" localizedCaseInsensitiveContainsString:@"اختبار ال"]);
     EXPECT_FALSE([@"TÉST" localizedCaseInsensitiveContainsString:@"Y"]);
     EXPECT_FALSE([@"TÉST" localizedCaseInsensitiveContainsString:@"TÉSTY"]);
     EXPECT_FALSE([@"TÉST" localizedCaseInsensitiveContainsString:@"TEST"]);
     EXPECT_FALSE([@"TÉST" localizedCaseInsensitiveContainsString:@"test"]);
 }
 
 TEST(NSString, LocalizedStandardContainsString) {
+    EXPECT_FALSE([@"" localizedStandardContainsString:nil]);
     EXPECT_FALSE([@"" localizedStandardContainsString:@""]);
     EXPECT_FALSE([@"TÉST" localizedStandardContainsString:@""]);
     EXPECT_TRUE([@"TÉST" localizedStandardContainsString:@"T"]);
     EXPECT_TRUE([@"TÉST" localizedStandardContainsString:@"TÉST"]);
+    EXPECT_TRUE([@"TE\u0301ST" localizedStandardContainsString:@"TÉST"]);
+    EXPECT_TRUE([@"TÉST" localizedStandardContainsString:@"TE\u0301ST"]);
```

Comment:
`localizedStandardContainsString:` should be case- AND diacritic-insensitive; can you test `TÉST` with its fully lowercase version, and also `te\u0301st` (which will test both diacritic sensitivity and case sensitivity?)